### ISSUE TITLE
sublist3r.py:614: DeprecationWarning:

### ISSUE THREAD

**Issue by Mr-Yacin** [aboul3la/Sublist3r#256]
sublist3r.py:614: DeprecationWarning: please use dns.resolver.Resolver.resolve() instead
  ip = Resolver.query(host, 'A')[0].to_text()

**Comment by Toriqulislamshimanto** [comment]
same problem for me as well

**Comment by xabix** [comment]
I have the same issue. Any idea? or update planned to fix this?

**Comment by aldisaw** [comment]
me too, any update ?

**Comment by Hell-yeahhz** [comment]
same here... Has anyone fixed it?

**Comment by ashish99039** [comment]
same here...anyone found the solution yet?

**Comment by RoninNakomoto** [comment]
I wrote a fix for this and submitted a pull request for it just now.
It is under:
#292

**Comment by ashish99039** [comment]
I have python 3+ but still getting the error. I am new here so if I am
missing out on something, do let me know

**Comment by RoninNakomoto** [comment]
@ashish99039 yes withdrawing the pull request. Going to test previous commits by others that might resolve the issue and will get back on that

**Comment by RoninNakomoto** [comment]
I will resubmit the fix specific to deprecated warning. I missed one additional deprecated statement.

**Comment by ashish99039** [comment]
Okay! Thanks a lot. Will be waiting

**Comment by diegorobalino16** [comment]
Solve this problem.
edit this file on the line:613 (ip = Resolver.query(host, 'A')[0].to_text())
/usr/lib/python3/dist-packages/sublist3r.py
put this line
ip = Resolver.**resolve**(host, 'A')[0].to_text()  
Save and Solve

**Comment by RoninNakomoto** [comment]
@ashish99039 @diegorobalino16 
check sublist3r2
https://github.com/RoninNakomoto/Sublist3r2

it works on python3, is does much faster bruteforcing. it is based on sublist3r.

**Comment by ashish99039** [comment]
> @ashish99039 @diegorobalino16 
> check sublist3r2
> https://github.com/RoninNakomoto/Sublist3r2
> 
> it works on python3, is does much faster bruteforcing. it is based on sublist3r.

Thanks so much. Will definitely check that

**Comment by detoober** [comment]
@ashish99039 
had same issue, edited the line mentioned in the error using nano editor and it worked fine after that. Alternatively you can try sublit3r2 also.

sudo nano sublist3r.py      
change the line as below
lease use dns.resolver.Resolver.resolve() instead
  ip = Resolver.query(host, 'A')[0].to_text()
save
and you are good to go
